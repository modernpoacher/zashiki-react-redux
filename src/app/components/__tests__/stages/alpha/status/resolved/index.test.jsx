import React from 'react'
import PropTypes from 'prop-types'

import {
  toSnapshot
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Resolved from '#zashiki-react-redux/app/components/stages/alpha/status/resolved'

jest.mock('react-redux', () => {
  return {
    connect () {
      return (Component) => Component
    }
  }
})

jest.mock('#zashiki-react-redux/app/router/with-router', () => (Component) => Component)

/**
 *  @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} props
 *  @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#zashiki-react-redux/app/components/stages/alpha/status/resolved', () => {
  const MOCK_DEFINITION = {
    meta: {
      uri: '#/'
    },
    elements: {
      title: 'MOCK TITLE',
      description: 'MOCK DESCRIPTION',
      field: {
        id: 'MOCK ID'
      }
    }
  }

  const MOCK_RESOURCE = {
    alpha: 'MOCK ALPHA',
    omega: 'MOCK OMEGA'
  }

  const MOCK_RESPONSE = {
    '#/': 'MOCK VALUE'
  }

  const MOCK_ERRORS = []

  const MOCK_DEFINITIONS = [
    {
      definition: MOCK_DEFINITION,
      resource: MOCK_RESOURCE,
      response: MOCK_RESPONSE,
      errors: MOCK_ERRORS
    }
  ]

  describe('Always', () => {
    it('renders', () => {
      expect(toSnapshot(render(
        <Resolved
          definitions={MOCK_DEFINITIONS}
          onChange={jest.fn()}
          onSubmit={jest.fn()}
        />
      )))
        .toMatchSnapshot()
    })
  })
})
