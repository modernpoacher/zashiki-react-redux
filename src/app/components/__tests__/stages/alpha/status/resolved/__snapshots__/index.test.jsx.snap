// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#zashiki-react-redux/app/components/stages/alpha/status/resolved Always renders 1`] = `
<div
  className="alpha resolved"
>
  <ol>
    <li>
      <form
        action="/MOCK ALPHA/MOCK OMEGA"
        method="POST"
        onSubmit={[Function]}
      >
        <h2 />
        <div
          className="cog text"
        >
          <label
            className="title text"
            htmlFor="MOCK ID"
          >
            <span
              className="text-content"
            >
              MOCK TITLE
            </span>
          </label>
          <input
            className="input text"
            disabled={false}
            id="MOCK ID"
            name="MOCK ID"
            onChange={[Function]}
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Continue
        </button>
      </form>
    </li>
  </ol>
</div>
`;
